doc/debugging.rst

```diff
@@ -14,33 +14,91 @@ This allows breakpoints and the execution flow to be viewed and controlled right
 Debugging with ipdb
 ===================
 
-Spyder offers the following debugging features integrated into the native GUI:
+You can fully control debugger execution from the :guilabel:`Debug` menu, :guilabel:`Debug toolbar` and via configurable keyboard shortcuts, along with the standard ``ipdb`` `console commands`_.
+Additionally, the  :doc:`editor` shows the line of code the debugger is currently stopped on with an arrow.
 
-* Multiple means of setting and clearing normal and conditional breakpoints for any line in a file opened in the :doc:`editor`.
+.. _console commands: https://pythonconquerstheuniverse.wordpress.com/2009/09/10/debugging-in-python/
 
-  * By selecting the respective option from the Debug menu.
-  * Through pressing a configurable keyboard shortcut (:kbd:`F12` for normal, or :kbd:`Shift-F12` for conditional breakpoints by default).
-  * By double-clicking to the left of the line number in an open file.
-  * With an ``ipdb.set_trace()`` statement in your code (after ``import`` ing ``pdb``).
-  * Interactively, using the ``b`` command in an ``ipdb`` session.
+GIF: Inset of Spyder's Editor, with a breakpoint set and the condition dialog open (commands from the toolbar)
+
+Spyder's debugger offers syntax highlighting, code completion and command history, which work exactly like they do in the normal interactive interpreter.
+Use the up and down arrows to recall previous commands, and press :kbd:`Tab` to trigger autocomplete suggestions.
+
+GIF: Browsing autocomplete suggestions
+
+Furthermore, IPython's `magic functions`_ are available in debugging mode.
+You can, for example, run ``%matplotlib <BACKEND>`` to switch between the ``qt5`` and ``inline`` plotting backends, or ``%timeit`` to check how fast a given snippet of code is.
+
+.. _magic functions: https://ipython.readthedocs.io/en/stable/interactive/magics.html
+
+SCREENSHOT: (timeit)
+
+SCREENSHOT of highlighting step (close up)
+
+Finally, you can enter and execute multiline statements in Spyder's debugger just like with the regular IPython prompt, to easily execute complex code while debugging.
```

Comment:
```suggestion
Finally, you can enter and execute multiline statements in Spyder's debugger just like with the regular IPython console, to easily execute complex code.
```

* Repeating again `IPython console` for familiarity
* Removing `while debugging` because you mentioned before that those commands are entered in `Spyder's debugger`.

Reply:
> Repeating again IPython console for familiarity

I changed this for the same reason I mentioned above. They are still _in_ the IPython console (the Spyder pane, as the both the UI and these docs consistently use to refer to it), just not using the normal IPython REPL, so it doesn't make sense to say just "like" it. However, if you have a better word than "prompt" (or "interpreter", REPL, etc) I'm all ears.

For what its worth, this would have been another good reason to simplify the name to just the `Console`, but we already fought that battle...

> Removing while debugging because you mentioned before that those commands are entered in Spyder's debugger.

Yeah, good point; I also noticed a repetition of `execute`. So, we thus have

```suggestion
Finally, you can enter and execute multiline statements in Spyder's debugger just like with the regular IPython prompt, to easily run complex code.
```